src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -302,7 +305,23 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary = False):
                 logging.debug("Updated job summary state history for job %s" % jobSummaryId)
 
             if job.get("fwjr", None):
-
+                cached_task = self.task_cache.get(job['task'])
+                if cached_task is None:
+                    spec_path = self.getWorkflowSpecDAO.execute(job['task'])['spec']
```

Comment:
The spec file is the same for the whole workflow (and tasks).
I wonder whether `job["task"]` is the whole task path, e.g.: /workflowName/taskPath or just the taskname? Do you have an example?

If it's the whole taskPath, then you could add it to the cache. If it's the taskName only, then it's a no-go because we have diff wfs with the same taskname.